### DISCUSSION TITLE
Add option to only print out t.log() if the test fails.

### DISCUSSION THREAD

**Discussion by dkstrong** [avajs/ava#3019] [Ideas]
* What you're trying to do
I am usually only interested in seeing logged items via t.log() if the test has failed (and i'm trying to debug/figure out why). I usually don't care about t.log for tests that pass, and find that it clutters up the test result output, making it harder to jump to and find the test results I do care about (the failing ones)

* Why you can't use AVA for this
There does not seem to be an ava configuration parameter or any other means to conditionally t.log() on a test failure.

* And maybe how you think AVA could handle this
Ideally there could be an argument added to the command line that is something like --displayLogOnlyOnFailedTests and/or a way to configure this in the "ava" section of package.json

**Comment by novemberborn** [comment]
Could you elaborate on your use of logs? Do you pepper them throughout in case a test starts to fail?

Personally I use them for debugging and sporadically to output say paths to temporary files generated by the test, so I can inspect those if needed.

**Comment by tommy-mitchell** [comment]
I recently used this snippet to only log information about a test if it fails, keeping the test output less noisy on successful runs:

```js
const actual = doSomething(input);

const assertions = await t.try(tt => {
	tt.is(actual, expected);
});

if (!assertions.passed) {
	t.log("input:", input);
}
```

**Reply by tommy-mitchell** [reply]
Since `t.log` mirrors `console.log` (in that it allows any number of arguments), maybe a new `t.logOnFail` could be added? My test could then be

```js
t.is(foo(input), expected);

t.logOnFail("input:", input);
```

---

For example, [here](https://github.com/tommy-mitchell/np/blob/784b2cdaec66ba8e859accc5373fe18e20e71fc5/test/_helpers/mock-inquirer.js) I'm mocking [`inquirer`](https://github.com/SBoudrias/Inquirer.js) using a custom answering system. I have a bunch of logs spread around for inspecting. If I wanted to log them with AVA (without using the above pattern), I'd have to comment/uncomment them manually.

**Reply by novemberborn** [reply]
Sorry for the late rely @tommy-mitchell.

Do we need log levels? Maybe not the full suite but I can see a `t.log.debug()` which always writes, and some rules for other levels where they're only on failure.

**Reply by tommy-mitchell** [reply]
Is `t.log.debug()` not the same as `t.log()`? And are there any cases besides just logging when the test fails?

**Reply by novemberborn** [reply]
Maybe log is failure only, and then the others will show depending on the log level. But at the end of the test run we say something about logs being hidden and provide the log level configuration needed to show them.

We may not need anything but debug to start with.
